ios/Voice/Voice.m

```diff
@@ -67,7 +68,12 @@ - (void) setupAndStartRecognizing:(NSString*)localeStr {
 
     // A recognition task represents a speech recognition session.
     // We keep a reference to the task so that it can be cancelled.
+    NSString *taskSessionId = self.sessionId;
     self.recognitionTask = [self.speechRecognizer recognitionTaskWithRequest:self.recognitionRequest resultHandler:^(SFSpeechRecognitionResult * _Nullable result, NSError * _Nullable error) {
+        if (![taskSessionId isEqualToString:self.sessionId]) {
```

Comment:
I have a question. Is it possible that this handler gets called with a different `sessionId`? Can you provide an input to produce this situation?

Do you think this method could get called even after `[self.recognitionTask cancel]` or `[self.recognitionTask finish] has been called?

Reply:
yes, in our testing, this method will get called even after [self.recognitionTask cancel] is called on in flight recognition that is already in progress

Reply:
To your first question, if you call start right after tearDown, you can easily reproduce this issue where this block is invoked on old session Id when a new session is generated

Reply:
Oh, thanks for your response! i think all your comments make sense!